src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Action } from 'vs/base/common/actions';
+import { localize } from 'vs/nls';
+import { INotebookService, INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import { NotebookComponent } from 'sql/workbench/contrib/notebook/browser/notebook.component';
+import { ICellModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { CellType } from 'sql/workbench/services/notebook/common/contracts';
+import { ToggleableAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
+
+export class CellToolbarAction extends Action {
+
+	constructor(
+		id: string,
+		label: string,
+		cssClass: string,
+		tooltip: string,
+		private _cellModel: ICellModel,
+		@INotebookService private _notebookService: INotebookService
+	) {
+		super(id, label, cssClass);
+		this._tooltip = tooltip;
+	}
+	// Todo: Add actions references here?
+}
+
+export class AddCellAction extends Action {
```

Comment:
This _should_ add a cell after current cell by default but no cell is being added when I click + > Markdown cell.
I borrowed this code block from notebookActions.ts

Reply:
Why are you redefining it here? If it's the same logic just use the already defined one.

Did you verify that this action is actually being called? It looks fine so you'll need to debug it a bit and provide some more information about where exactly it's failing.

Reply:
Agreed; if we can leverage an existing action, let's do that. If it's moving from the ... into here, then let's move the action itself from cellToggleMoreActions.ts to this file.